pkg/apis/build/v1alpha2/build_pod.go

```diff
@@ -0,0 +1,652 @@
+package v1alpha2
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/kmeta"
+
+	"github.com/pivotal/kpack/pkg/apis/build/v1alpha1"
+)
+
+const (
+	directExecute   = "--"
+	buildInitBinary = "build-init"
+	rebaseBinary    = "rebase"
+
+	SecretTemplateName           = "secret-volume-%s"
+	SecretPathName               = "/var/build-secrets/%s"
+	BuildLabel                   = "kpack.io/build"
+	DOCKERSecretAnnotationPrefix = "kpack.io/docker"
+	GITSecretAnnotationPrefix    = "kpack.io/git"
+
+	cacheDirName              = "cache-dir"
+	layersDirName             = "layers-dir"
+	platformDir               = "platform-dir"
+	homeDir                   = "home-dir"
+	workspaceDir              = "workspace-dir"
+	imagePullSecretsDirName   = "image-pull-secrets-dir"
+	builderPullSecretsDirName = "builder-pull-secrets-dir"
+
+	serviceBindingRootEnvVar = "SERVICE_BINDING_ROOT"
+)
+
+type BuildPodImages struct {
+	BuildInitImage  string
+	CompletionImage string
+	RebaseImage     string
+}
+
+type BuildPodBuilderConfig struct {
+	StackID     string
+	RunImage    string
+	Uid         int64
+	Gid         int64
+	PlatformAPI string
+}
+
+type ServiceBinding struct {
+	Name                string
+	SecretRef           *corev1.LocalObjectReference
+	V1Alpha1MetadataRef *corev1.LocalObjectReference
+}
+
+var (
+	sourceVolume = corev1.VolumeMount{
+		Name:      workspaceDir,
+		MountPath: "/workspace",
+	}
+	homeVolume = corev1.VolumeMount{
+		Name:      homeDir,
+		MountPath: "/builder/home",
+	}
+	platformVolume = corev1.VolumeMount{
+		Name:      platformDir,
+		MountPath: "/platform",
+	}
+	cacheVolume = corev1.VolumeMount{
+		Name:      cacheDirName,
+		MountPath: "/cache",
+	}
+	layersVolume = corev1.VolumeMount{
+		Name:      layersDirName,
+		MountPath: "/layers",
+	}
+	projectMetadataVolume = corev1.VolumeMount{
+		Name:      layersDirName,
+		MountPath: "/projectMetadata",
+	}
+	homeEnv = corev1.EnvVar{
+		Name:  "HOME",
+		Value: "/builder/home",
+	}
+	imagePullSecretsVolume = corev1.VolumeMount{
+		Name:      imagePullSecretsDirName,
+		MountPath: "/imagePullSecrets",
+		ReadOnly:  true,
+	}
+	builderPullSecretsVolume = corev1.VolumeMount{
+		Name:      builderPullSecretsDirName,
+		MountPath: "/builderPullSecrets",
+		ReadOnly:  true,
+	}
+	serviceBindingRootEnv = corev1.EnvVar{
+		Name:  serviceBindingRootEnvVar,
+		Value: filepath.Join(platformVolume.MountPath, "bindings"),
```

Comment:
K8s service bindings and cnb service bindings should not share the same directory.

Reply:
As I understand it, buildpacks will only read from one location for bindings based on [this]( https://github.com/buildpacks/libcnb/blob/main/build.go#L166) and a quick convo with @nebhale 

Given that, I made build pods only project CNB bindings OR K8s bindings in which case why not use the same directory?

Curious to peoples' thoughts

Reply:
@tylerphelan that makes sense to me. I always assumed it would be either/or and not both.